### PR TITLE
Improve the serializer

### PR DESCRIPTION
**Pull request by BenediktBurger**
Adding comments, type hints and an enum for consistent naming (you can call enum members).

Also makes the serializer symmetrical to the deserializer.

Fixes that a boolean is serialized as an integer, as the number serialization ran before the boolean test. (`isinstance(True, int) returns True`).

### CODE REVIEW

File: src/pymodaq/utils/tcp_ip/serializer.py [hunk 1]
```diff
@@ -20,17 +21,38 @@
 
 
 SERIALIZABLE = Union[
-    int,
+    # native
+    bool,
+    bytes,
     str,
-    numbers.Number,
+    complex,  # float and int are subtypes for type hinting
```
Comment (seb5g) [inline src/pymodaq/utils/tcp_ip/serializer.py:28]: ok

File: src/pymodaq/utils/tcp_ip/serializer.py [hunk 2]
```diff
@@ -205,18 +227,19 @@ def string_serialization(self, string: str) -> bytes:
...
         ----------
-        scalar: str
+        scalar: A python number (complex or subtypes like float and int)
 
         Returns
         -------
         bytes: the total bytes message to serialize the scalar
         """
         if not isinstance(scalar, numbers.Number):
+            # type hint is complex, instance comparison Number
```
Comment (seb5g) [inline src/pymodaq/utils/tcp_ip/serializer.py:242]: ok, then is not numbers.Numbers also valid for complex?

File: src/pymodaq/utils/tcp_ip/serializer.py [hunk 3]
```diff
@@ -350,49 +373,56 @@ def list_serialization(self, list_object: List) -> bytes:
...
             bytes_string += self.list_serialization(obj)
 
         elif isinstance(obj, putils.ParameterWithPath):
             path = obj.path
             param_as_xml = ioxml.parameter_to_xml_string(obj.parameter)
-            bytes_string += self.string_serialization('parameter')
+            bytes_string += self.string_serialization(SerializableTypes.PARAMETER.value)
             bytes_string += self.list_serialization(path)
-            bytes_string += self.string_serialization(param_as_xml)
+            bytes_string += self.bytes_serialization(param_as_xml)
```
Comment (seb5g) [inline src/pymodaq/utils/tcp_ip/serializer.py:422]: indeed it is a bynary string

**Review by seb5g — APPROVED** [review]

File: src/pymodaq/utils/tcp_ip/serializer.py (hunk 2 above)
Comment (BenediktBurger) [reply src/pymodaq/utils/tcp_ip/serializer.py:242]: The most permissive type hint is complex (allowing also float and int).
For instance comparison, you have to use numbers.Number, which will return true for complex, float, or int.

Comment (seb5g) [reply src/pymodaq/utils/tcp_ip/serializer.py:242]: ok then

**Pull request merged**